File: programs/group-anagram.cs
public class Solution {
    public IList<IList<string>> GroupAnagrams(string[] strs) {
        var myDict = new Dictionary<string,IList<string>>();

        foreach(string str in strs){
            int[] count = new int[26];
            foreach(char c in str){
                count[c - 'a']++;
            }
            string myKey = string.Join("#",count);
            if(!myDict.ContainsKey(myKey)){
                myDict[myKey] = new List<string>();
            }
            myDict[myKey].Add(str);
        }
        return new List<IList<string>>(myDict.Values);
    }
}

File: programs in python/group-anagram.py
from collections import defaultdict
from typing import List
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        mydict = defaultdict(list)

        for s in strs:
            count = [0]*26
            for i in s:
                count[ord(i) - ord("a")] += 1
            
            key = tuple(count)
            mydict[key].append(s)
        
        return mydict.values()
            